### PR TITLE
Raise error when function as llm_config passed to GroupChatManager

### PR DESCRIPTION
**Pull request by kevin666aa**
1. Raise error when function as llm_config passed to GroupChatManager
2. Add test in `test_function_call_groupchat.py`.
3. Handling undesired case when selector is returning a function_call instead of str name. 

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: autogen/agentchat/groupchat.py [hunk 1]
```diff
@@ -217,6 +217,20 @@ def _mentioned_agents(self, message_content: Union[str, List], agents: List[Agen
...
+        if isinstance(message_content, dict):
+            tmp_content = ""
+            if message_content.get("content"):
+                tmp_content = message_content["content"]
+            if message_content.get("function_call"):
+                tmp_content += message_content["function_call"]["name"] + message_content["function_call"]["arguments"]
+            if message_content.get("tool_calls"):
+                for tool_call in message_content["tool_calls"]:
+                    if tool_call.get("type") == "function":
+                        tmp_content += tool_call["function"]["name"] + tool_call["function_call"]["arguments"]
```
Comment (sonichi) [inline autogen/agentchat/groupchat.py:231]: tmp_content is not used afterwards. Do we still need this part after disabling tool/function calls for group chat manager?

File: test/agentchat/test_function_call_groupchat.py [hunk 2]
```diff
@@ -54,7 +54,19 @@ def get_random_number():
...
     groupchat = autogen.GroupChat(agents=[user_proxy, coder], messages=[], max_round=7)
-    manager = autogen.GroupChatManager(groupchat=groupchat, llm_config=llm_config)
+
+    # pass in llm_config with functions
+    try:
+        manager = autogen.GroupChatManager(groupchat=groupchat, llm_config=llm_config)
+    except ValueError as e:
+        print("ValueError is expected: ", e)
+    else:
+        raise ValueError("Should raise ValueError because of 'functions' passed to GroupChatManager.")
```
Comment (sonichi) [inline test/agentchat/test_function_call_groupchat.py:64]: use pytest.assertError() as used in some other tests.

File: autogen/agentchat/groupchat.py (hunk 1 above)
Comment (kevin666aa) [reply autogen/agentchat/groupchat.py:231]: My mistake

**Comment by afourney** [comment]
Sounds good. Have we fixed that notebook?

**Comment by kevin666aa** [comment]
> Sounds good. Have we fixed that notebook?

Right. Thanks for the heads up. Let me update it.

**Comment by kevin666aa** [comment]
@afourney I went through the notebooks and fix one in `agentchat_groupchat_RAG.ipynb`. Am I missing anything? I didn't find other notebooks that are passing functions to groupmanager.

**Comment by sonichi** [comment]
Test fails: https://github.com/microsoft/autogen/actions/runs/7137474617/job/19437525044?pr=911

File: autogen/agentchat/groupchat.py [hunk 3]
```diff
@@ -259,6 +259,20 @@ def _mentioned_agents(self, message_content: Union[str, List], agents: List[Agen
...
```
Comment (sonichi) [inline autogen/agentchat/groupchat.py:273]: Is this unnecessary if functions and tools are disabled?

Comment (kevin666aa) [reply autogen/agentchat/groupchat.py:273]: Yes, I am removing that

**Pull request merged**